### PR TITLE
#541 Extension objects pattern

### PR DESCRIPTION
**Pull request by SrdjanPaunovic**
Pull request for issue #541

### CODE REVIEW

File: execute-around/pom.xml [hunk 1]
```diff
@@ -23,7 +23,9 @@
     THE SOFTWARE.
 
 -->
-<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0
+    http://maven.apache.org/xsd/maven-4.0.0.xsd"
+         xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (iluwatar) [inline execute-around/pom.xml]: Unnecessary change, this file belongs to another pattern

File: extension-objects/src/main/java/App.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import abstractextensions.CommanderExtension;
+import abstractextensions.SergeantExtension;
+import abstractextensions.SoldierExtension;
+import units.CommanderUnit;
+import units.SergeantUnit;
+import units.SoldierUnit;
+import units.Unit;
+
+/**
+ * Created by Srdjan on 26-Apr-17.
```
Comment (iluwatar) [inline extension-objects/src/main/java/App.java]: Please create meaningful JavaDocs instead of author names

File: extension-objects/src/main/java/App.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+   * Program entry point
+   *
+   * @param args command line args
+   */
+  public static void main(String[] args) {
+
+    //Create 3 different units
+    Unit unit = new SoldierUnit("SoldierUnit1");
+    Unit unit1 = new SergeantUnit("SergeantUnit1");
+    Unit unit2 = new CommanderUnit("CommanderUnit1");
```
Comment (iluwatar) [inline extension-objects/src/main/java/App.java]: Instead of unit, unit1 and unit2 how about soldierUnit, sergeantUnit and commanderUnit?

File: extension-objects/src/main/java/App.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    //check for each unit to have an extension
+    checkExtensionsForUnit(unit);
+    checkExtensionsForUnit(unit1);
+    checkExtensionsForUnit(unit2);
+
+  }
+
+  private static void checkExtensionsForUnit(Unit unit) {
+    //separate for better view
+    System.out.println();
```
Comment (iluwatar) [inline extension-objects/src/main/java/App.java]: Please use a logging framework instead of `System.out.println`

**Comment by iluwatar** [comment]
@SrdjanPaunovic you have my review comments. Please notify when the changes are done.

**Comment by SrdjanPaunovic** [comment]
Notify @iluwatar

**Comment by iluwatar** [comment]
Looks good @SrdjanPaunovic thank you for the pattern contribution! :+1:

**Comment by iluwatar** [comment]
@all-contributors please add @SrdjanPaunovic for code
